inference/enterprise/device_manager/pubsub.py

```diff
@@ -0,0 +1,96 @@
+import json
+import time
+import uuid
+
+from paho.mqtt import client as mqtt_client
+
+from inference.core.env import (
+    API_KEY,
+    DEVICE_MANAGER_PASSWORD,
+    DEVICE_MANAGER_PUBSUB_HOST,
+    DEVICE_MANAGER_USERNAME,
+)
+from inference.core.logger import logger
+from inference.enterprise.device_manager.command_handler import handle_command
+from inference.enterprise.device_manager.helpers import get_device_id
+
+METRICS_TOPIC = f"roboflow/device-management/v1/{API_KEY}/metrics"
+COMMANDS_TOPIC = f"roboflow/device-management/v1/{API_KEY}/commands"
+STREAM_TOPIC = f"roboflow/device-management/v1/{API_KEY}/stream"
+SYS_TOPIC = f"roboflow/device-management/v1/{API_KEY}/system"
+
+MAX_RECONNECT_COUNT = 12
+RECONNECT_RATE = 2
+MAX_RECONNECT_DELAY = 60
+
+CLIENT_ID = f"device-managed-{hex(uuid.getnode())}"
+
+
+def on_connect(client, userdata, flags, rc):
+    if rc == 0 and client.is_connected():
+        logger.info("Connection established")
+        client.subscribe(COMMANDS_TOPIC)
+    else:
+        logger.error("Failed to connect to messaging broker, return code %d\n" % rc)
+
+
+def on_disconnect(client, userdata, rc):
+    reconnect_count, reconnect_delay = 0, 1
+    while reconnect_count < MAX_RECONNECT_COUNT:
+        time.sleep(reconnect_delay)
+        try:
+            client.reconnect()
```

Comment:
is it possible for the client to have become connected (`on_connect` called) while we were asleep? I'm wondering if we should check here to see if we still need to `reconnect`.